spec/firebase_app_distribution_auth_client_spec.rb

```diff
@@ -0,0 +1,84 @@
+describe Fastlane::Auth::FirebaseAppDistributionAuthClient do
+  let(:auth_helper) { Class.new { extend(Fastlane::Auth::FirebaseAppDistributionAuthClient) } }
+  let(:fake_binary) { double("Binary") }
+  let(:fake_binary_contents) { double("Contents") }
+  let(:auth) { Google::Auth::ServiceAccountCredentials }
+  let(:fake_service_creds) { double("service_account_creds") }
+  let(:fake_auth_client) { double("auth_client") }
+  let(:payload) { { "access_token" => "service_fake_auth_token" } }
+
+  before(:each) do
+    allow(Signet::OAuth2::Client).to receive(:new)
+      .and_return(fake_auth_client)
+    allow(fake_auth_client).to receive(:fetch_access_token!)
+    allow(fake_auth_client).to receive(:access_token)
+      .and_return("fake_auth_token")
+
+    allow(auth).to receive(:make_creds)
+      .with(anything)
+      .and_return(fake_service_creds)
+    allow(fake_service_creds).to receive(:fetch_access_token!)
+      .and_return(payload)
+
+    allow(File).to receive(:open)
+      .and_return(fake_binary)
+    allow(fake_binary).to receive(:read)
+      .and_return(fake_binary_contents)
+    allow(fake_binary_contents).to receive(:key)
+      .and_return("fake_service_key")
+
+    allow(ENV).to receive(:[])
+      .with("GOOGLE_APPLICATION_CREDENTIALS")
+      .and_return(nil)
+  end
+
+  describe '#fetch_auth_token' do
+    it 'uses service credentials for authorization when the path is passed in' do
+      expect(auth_helper.fetch_auth_token("google_service_path")).to eq("service_fake_auth_token")
+    end
+
+    it 'uses service credentials for authorization when the enviormental variable is set' do
+      allow(ENV).to receive(:[])
+        .with("GOOGLE_APPLICATION_CREDENTIALS")
+        .and_return("google_service_path")
+      expect(auth_helper.fetch_auth_token("")).to eq("service_fake_auth_token")
+    end
+
+    it 'uses service credentials for authorization when the enviormental variable is set and path is nil' do
+      allow(ENV).to receive(:[])
```

Comment:
Consider changing these `allow(ENV)` to `expect(ENV)` to make sure that they are being called.